### ISSUE TITLE
Error on Bulk operation

### ISSUE THREAD

**Issue by bjoernHeneka** [akeneo/pim-community-dev#7467]
Currently i've got a problem with bulk operation.
I'm using the current CE Version 2.0.11

I'm selecting about 1000 ofproducst for a Bulk operation like "remove from category" When starting the job i got the following error after 200 processed entities.

```
FAILED
A new entity was found through the relationship 'Pim\Component\Catalog\Model\AbstractProduct#family' that was not configured to cascade persist operations for entity: [hardDisk]. To solve this issue: Either explicitly call EntityManager#persist() on this unknown entity or configure cascade persist this association in the mapping for example @ManyToOne(..,cascade={"persist"}).
```

![bulk_error](https://user-images.githubusercontent.com/10229990/34725901-aac8b866-f552-11e7-9264-784a55ccb2c0.png)

**Comment by triada63** [comment]
I have same problem, but when mass add to categories, after 200 rows i get this error.

**Comment by dng-dev** [comment]
We have the Same issue after mass updating and assigning products to families

**Comment by nowiko** [comment]
@bjoernHeneka  @triada63 @dng-dev 
It seems like this case is related to `UnitOfWorkAndRepositoriesClearer->clear()`. 
While processing the first bulk of entities we clear entity manager and that is why next batch of entities points to the detached related family entity. (actually, it fails after 100 rows processed and written).
I have commented out this method and it works, but not sure how this might impact on performance and bulk operation in general.

Could somebody from core developers assist on this?

**Comment by nowiko** [comment]
@jjanvier could you please pay attention to this issue? As it seems like it influence on usage of bulk operations in general.

**Comment by LaureBro** [comment]
Hello, 
We have fixed issues on mass bulk actions, do you still have these problems on our latest versions?
Kind regards, 
Laure

**Comment by nei** [comment]
I am using version 2.1.0 CE and got this error when using "Mass remove from categories" bulk action.
```
A new entity was found through the relationship 'Pim\Component\Catalog\Model\AbstractProduct#family' that was not configured to cascade persist operations for entity: [Tools]. To solve this issue: Either explicitly call EntityManager#persist() on this unknown entity or configure cascade persist this association in the mapping for example @ManyToOne(..,cascade={"persist"}).
```
In which version was this fixed? @LaureBro I looked the changelog and I couldnt figure it out. if you let me know which version I will give it a try upgrading it to latest.

**Comment by LaureBro** [comment]
Hello @nei, 

If I am correct our internal reference is PIM-7131 (matching this GitHub PR #7565) merged in 2.0.14. So it should fixed on 2.1.4 and above, but the best is to update on the latest patch, the 2.1.8.

**Comment by nei** [comment]
Thank you @LaureBro I will update to latest.

**Comment by marckvandijk** [comment]
We are facing the same issue. I believe we have the latest version running (version 2.3.8). In previous posts I see that it might has something to do with the numbers of rows. But the Excel file I want to import only has 17 rows. 

Could it be a bug or does someone have an idea what I might could try?

Marck

**Comment by Doodoune** [comment]
I close this issue, feel free to re-open it if you still have the issue and don't hesitate to provide us step by step explanation to reproduce the problem & the value of your parameter "pim_job_product_batch_size" (if you override it).

Regards,

**Comment by akosglue** [comment]
same issue in 2.2.12

**Comment by triada63** [comment]
I got same error in my custom code, it's throwing when you loading model of products, then work with other model, persist and after save products that been loaded. Next batch throws error, i don't know what it mean's, but something wrong with relations, helps to clear only all used repositories on every batch, then next batch without problems.

**Comment by akosglue** [comment]
same issue in 3.0.36

**Comment by flagbird** [comment]
It is still with 3.0.46.

**Comment by flagbird** [comment]
It is supposed to be fixed since 3.0.47.
